src/Sentry.Azure.Functions.Worker/SentryFunctionsWorkerMiddleware.cs

```diff
@@ -94,7 +96,11 @@ private async Task<TransactionContext> StartOrContinueTraceAsync(FunctionContext
         var httpMethod = requestData.Method.ToUpperInvariant();
         var transactionNameKey = $"{context.FunctionDefinition.EntryPoint}-{httpMethod}";
 
-        if (!TransactionNameCache.TryGetValue(transactionNameKey, out transactionName))
+        // Note that, when Trimming is enabled, we can't use reflection to read route data from the HttpTrigger
+        // attribute. In that case the route name will always be /api/<FUNCTION_NAME>
+        // If this is ever a problem for customers, we can potentially see if there are alternate ways to get this info
+        // from route tables or something. We're not even sure if anyone will use this functionality for now though. 
+        if (!AotHelper.IsNativeAot && !TransactionNameCache.TryGetValue(transactionNameKey, out transactionName))
```

Comment:
For this AotHelper, instead of:

https://github.com/getsentry/sentry-dotnet/blob/217c11cea611a5f26e4a1abfbb7df37f34695722/src/Sentry/Internal/AotHelper.cs#L18-L19

Could we instead write an attribute at compile time? I imagine it's a lot cheaper (worth measuring) than trying to get a stack trace on a static constructor:

Something like this:
https://github.com/getsentry/sentry-dotnet/blob/217c11cea611a5f26e4a1abfbb7df37f34695722/src/Sentry/buildTransitive/Sentry.targets#L8-L29


Here's a PR that added writing the attribute at build time as a reference: https://github.com/getsentry/sentry-dotnet/pull/1739

Reply:
Hm, we could look into that... I'll create a ticket from your comment.